onnxruntime/core/providers/nuphar/compiler/x86/scheduler/tensorize_schedule.cc

```diff
@@ -89,31 +90,54 @@ static Status TensorizeGEMVInteger(const tvm::Tensor& tensor,
   return Status::OK();
 }
 
-static Status TensorizeReduction(const tvm::Tensor& tensor,
-                                 tvm_codegen::ScheduleContext& ctx,
-                                 const std::string& target_str) {
+static Status TensorizeIGEMV(const tvm::Tensor& tensor,
+                             tvm_codegen::ScheduleContext& ctx,
+                             bool tensorize,
+                             const std::string& target_str) {
   // Schedule tensor and inputs as root
   bool status_imatmul = InsertRootScheduleAndClosure(tensor, ctx);
   if (status_imatmul == false)
     return Status::OK();
-
   InputRootScheduleWithVectorizationX86(tensor, ctx);
-  // Loop tiling
-  int reduce_tile_size = (target_str == "avx512-skylake") ? 1024 : 512;
+
+  // Default tiling size
+  // TODO: tuning tiling sizes later
+  int tensorize_embed = 1;
+  int tensorize_input = (target_str == "avx512-skylake") ? 1024 : (target_str == "avx2") ? 512 : 256;
+
+  // Tensorize kernel shape
+  std::vector<int32_t> kernel_shape;
+  kernel_shape.push_back(tensorize_embed);
+  kernel_shape.push_back(tensorize_input);
+
   auto compute_op = tensor->op.as<tvm::ComputeOpNode>();
+  auto xy = compute_op->axis;
+  auto x = xy[0];
+  auto y = xy[1];
   auto z = compute_op->reduce_axis[0];
+
+  // no tiling need for IterVar x
+  tvm::IterVar yo, yi;
+  ctx.schedule[tensor->op].split(y, kernel_shape[0], &yo, &yi);
   tvm::IterVar zo, zi;
-  ctx.schedule[tensor->op].split(z, reduce_tile_size, &zo, &zi);
+  ctx.schedule[tensor->op].split(z, kernel_shape[1], &zo, &zi);
+  ctx.schedule[tensor->op].reorder({x, yo, zo, yi, zi});
+
+  if (tensorize) {
+    // TODO: refine tensorize gemv class
+    TensorizeIntGemv8bit igemv8bit("igemv8bit", kernel_shape);
+    ctx.schedule[tensor->op].tensorize(yi, igemv8bit.CreateTensorIntrin());
+  }
   return Status::OK();
 }
 
-static Status TensorizeGEMMInteger(const tvm::Tensor& tensor,
-                                   tvm_codegen::CodeGenContext& ctx_codegen,
-                                   tvm_codegen::ScheduleContext& ctx,
-                                   tvm::Expr batchseq_expr,
-                                   const std::vector<int64_t> embed_dim_vec,
-                                   const std::vector<int64_t> input_dim_vec,
-                                   const std::string& target_str) {
+static Status TensorizeIGEMM(const tvm::Tensor& tensor,
+                             tvm_codegen::CodeGenContext& ctx_codegen,
+                             tvm_codegen::ScheduleContext& ctx,
+                             tvm::Expr batchseq_expr,
+                             const std::vector<int64_t> embed_dim_vec,
+                             const std::vector<int64_t> input_dim_vec,
+                             const std::string& target_str) {
```

Comment:
>const std::string& target_str [](start = 29, length = 29)

Not necessary for this PR, but can we move this into CodeGenContext, and replace string comparison to flag checks?